packages/app/src/app/overmind/effects/api.ts

```diff
@@ -1,43 +1,29 @@
+import axios, { AxiosResponse, AxiosError } from 'axios';
 import { logError } from '@codesandbox/common/lib/utils/analytics';
 import { values } from 'lodash-es';
 import { camelizeKeys, decamelizeKeys } from 'humps';
-// import { addNotification } from '../factories';
 
-/*
-  This effect needs to expose a "configure" method where jwt
-  and error action is passed in
-*/
 const API_ROOT = '/api/v1';
 
-function createHeaders({ state, jwt }) {
-  const foundJwt = state.get('jwt') || jwt.get();
-
-  return foundJwt
-    ? {
-        Authorization: `Bearer ${foundJwt}`,
-      }
-    : {};
-}
-
-const getMessage = (error: Error & { response?: any }) => {
+const getMessage = (error: AxiosError) => {
   const response = error.response;
 
   if (!response || response.status >= 500) {
     logError(error);
   }
 
-  if (response && response.result) {
-    if (response.result.errors) {
-      const errors = values(response.result.errors)[0];
+  if (response && response.data) {
```

Comment:
This if - else hell looks overcomplicated. Is it any way that you can simplify it?

Reply:
Ah, yeah, there are tons of places we want to improve logic. Though currently I have to focus on moving over to Typescript or it will never be done :) So first over to Typescript and Overmind, then when things are running we can do another round of refactor to improve stuff like this. 

I totally agree with you, I just know myself that if I start doing these tiny improvements I get totally lost in the main goal of getting over to Typescript and new state management :)